pkg/controllers/register.go

```diff
@@ -201,6 +211,33 @@ func RegisterControllers(ctx *context.ControllerContext, syncers []syncer.Object
 	return nil
 }
 
+func registerGenericSyncController(ctx *context.ControllerContext) error {
+	// first check if a generic CRD config is provided and we actually need
+	// to create any of these syncer controllers
+	c := os.Getenv(ctx.Options.GenericConfig)
+	if c == "---" {
+		// empty configuration, no need for creating any syncer controllers
+		klog.Info("no generic config provided, skipping creating controllers")
+
+		return nil
+	}
+
+	configuration, err := config.Parse(c)
+	if err != nil {
+		klog.Error("error parsing the config", err.Error())
+		return err
+	}
+
+	klog.Info("generic config provided, parsed successfully")
```

Comment:
nit-picks: we could avoid klog usage here by wrapping the error that we return and using `loghelper.Infof` for info msgs, as we already do in another place in this file.